contracts/DepositContract.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.4.22;
+
+contract DepositContract {
+
+  mapping(address => uint) private Stakes;
+  mapping(address => uint) private Indices;
+  address[] private nodeAddressList;
```

Comment:
Do we need this list? A single map is not enough?

Reply:
Any key is trivially "present" in a solidity map (i.e. `map[k] = default_value`). The list is useful to delete a `nodeAddress` and keep track of live stakes.

Reply:
We can not delete the key from the map? I assume we can. Then no need for a separate list, right?

Reply:
A key is always present in a map.